tests/test_middleware.py

```diff
@@ -28,13 +30,56 @@ def session_refresh(dummy_view):
     return SessionRefresh(dummy_view)
 
 
+@pytest.fixture()
+def request_factory(rf: RequestFactory, session_middleware):
+    def _factory(config_class: OpenIDConnectConfigBase | None = None):
```

Comment:
we tend to use pytest marks to pass parameters instead of returning factory functions via a fixture. See for example https://github.com/maykinmedia/mozilla-django-oidc-db/blob/master/tests/conftest.py#L90

It leads to more readable tests because configuration is applied as a decorator to the test rather than part of the test.